### PR TITLE
Remove deprecated direct DB access in favor of RPC API

### PR DESCRIPTION
**Pull request by thotypous**
This allows the script to run on Mitaka deployed by OpenStack-Ansible.

Direct DB access on compute nodes is currently deprecated. Deployments made by OpenStack-Ansible don't even include database access credentials in nova.conf. Our solution was to use the RPC API instead of directly accessing the database.

### CODE REVIEW

File: faulty_device_cleanup.py [hunk 1]
```diff
@@ -31,13 +33,20 @@
...
-from nova import context
-from nova.db.sqlalchemy import api as db_api
-from nova.db.sqlalchemy import models
+import nova.context
 from nova import utils
 
+from nova.conductor import rpcapi as conductor_rpcapi
+from nova import objects
+from nova.objects import base as objects_base
+from nova import utils
```
Comment (jay-tsu) [inline faulty_device_cleanup.py]: redefinition of 'utils' from line 39.

File: faulty_device_cleanup.py [hunk 2]
```diff
@@ -66,15 +75,13 @@ def _get_host_name(self):
...
         # Find the targets used by VM on the compute node
-        bdms = db_api.model_query(context.get_admin_context(),
-                                  models.BlockDeviceMapping,
-                                  session=db_api.get_session())
-        bdms = bdms.filter(models.BlockDeviceMapping.connection_info != None)
-        bdms = bdms.join(models.BlockDeviceMapping.instance).filter_by(
-            host=string.strip(self.host_name))
+        context = nova.context.get_admin_context()
+        host_name = string.strip(self.host_name)
+        instance_uuids = [inst.uuid for inst in objects.InstanceList.get_by_host(context, host_name)]
```
Comment (jay-tsu) [inline faulty_device_cleanup.py]: E501 line too long (101 > 79 characters)

File: faulty_device_cleanup.py [hunk 3]
```diff
@@ -66,15 +75,13 @@ def _get_host_name(self):
...
+        bdms = objects.BlockDeviceMappingList.get_by_instance_uuids(context, instance_uuids)
```
Comment (jay-tsu) [inline faulty_device_cleanup.py]: E501 line too long (101 > 79 characters)

**Comment by jay-tsu** [comment]
@thotypous Thanks a lot! The code change looks good to me. Please run pep8 check before submitting the code.  :)

**Comment by thotypous** [comment]
@jay-tsu Thank you very much for the review. I have just fixed the diff for this pull request according to your remarks.

**Pull request merged**